### DISCUSSION TITLE
How to trace income request header

### DISCUSSION THREAD

**Discussion by zuk22** [plausible/analytics#4761] [Self-Hosted Support] [answered]
Hi!
Is this code still works? I am unable to access bin/plausible remote

```console
$ cd hosting
$ docker compose exec plausible bin/plausible remote
```
and there you can add a tracer for the `RemoteIP` module to print the incoming headers
```elixir
iex> headers = fn {:trace, _pid, :call, {_mod, _fun, [%Plug.Conn{req_headers: headers}]}} -> inspect(headers) end
iex> mod_fun_args = {PlausibleWeb.RemoteIP, :get, 1}
iex> how_many_times = 5
iex> :recon_trace.calls(mod_fun_args, how_many_times, formatter: headers)
```
Example output for:
```sh
curl http://localhost:8000/api/event -d '{"name": "pageview","url":"https://dummy.site/baldur","domain":"dummy.site","referrer":null,"screen_width":520}' -H 'content-type: application/json'
```
```
[{"accept", "*/*"}, {"content-length", "111"}, {"content-type", "application/json"}, {"host", "localhost:8000"}, {"user-agent", "curl/7.86.0"}]
```

_Originally posted by @ruslandoga in https://github.com/plausible/analytics/discussions/2767#discussioncomment-5367417_

**Comment by ruslandoga** [comment]
👋 @zuk22 

Yes, it should still work.

> I am unable to access bin/plausible remote

Do you see any errors in the output?

**Reply by zuk22** [reply]
It shows **no configuration file provided: not found**

**Reply by ruslandoga** [reply, marked as answer]
You should run it from the directory with `compose.yml` file, i.e. the directory where you cloned `plausible/community-edition`

**Reply by zuk22** [reply]
You are right, sorry for disturbing you!
